### PR TITLE
feat(notification): #CRRE-506 add admin daily notification with new order count

### PR DESCRIPTION
**Pull request by Sofianebernoussi**
## Describe your changes

- Add a notification for admin to notify them new orders (CRON every day).

## Checklist tests

- Check if the CRON pass everyday
- Check if the admin has a notification with the right number of new orders.

## Issue ticket number and link
https://jira.support-ent.fr/browse/CRRE-506

## Checklist before requesting a review (magic string, indentation, comment/documentation...)

- [x] I have detailed the tests to do in my feature/fix in order to prevent consequents regressions (must specify in **Checklist tests**)
- [x] I have performed a self-review of my code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation (API Doc etc...) - (must specify in **Description** for target version)
- [ ] If it is a consequent feature, I have added thorough tests.
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] Any dependent changes have been added to this project (must specify in **Description**)

### CODE REVIEW

File: deployment/crre/conf.json.template [hunk 1]
```diff
@@ -28,6 +28,7 @@
          "timeSecondSynchCron": "${timeSecondSynchCron}",
          "timeSecondStatCron": "${timeSecondStatCron}",
          "timeSecondStatutCron": "${timeSecondStatutCron}",
+         "timeSecondNotifyAdminsCron": "${timeSecondNotifyAdminsCron}",
```
Comment (sjollois) [inline deployment/crre/conf.json.template:31]: Tu peux rajouter la doc dans le Read.me aussi stp

File: src/main/java/fr/openent/crre/service/impl/DefaultNotificationService.java [hunk 2]
```diff
@@ -100,6 +115,7 @@ public void sendNotificationValidator(List<Integer> orderRegionEquipmentList) {
     }
 
     @Override
+    @SuppressWarnings("unchecked")
```
Comment (sjollois) [inline src/main/java/fr/openent/crre/service/impl/DefaultNotificationService.java:118]: ça sert à quoi de rajouter ça ?

**Review by sjollois — APPROVED** [review]

File: src/main/java/fr/openent/crre/cron/notifyNewOrdersAdmin.java [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+import fr.openent.crre.service.ServiceFactory;
+import fr.openent.crre.service.impl.DefaultNotificationService;
+import fr.openent.crre.service.impl.DefaultStructureService;
+import fr.wseduc.webutils.Either;
+import io.vertx.core.Handler;
+import io.vertx.core.json.JsonArray;
+import io.vertx.core.json.JsonObject;
+import org.entcore.common.controller.ControllerHelper;
+
+public class notifyNewOrdersAdmin extends ControllerHelper implements Handler<Long> {
```
Comment (BastienBenard) [inline src/main/java/fr/openent/crre/cron/notifyNewOrdersAdmin.java:12]: NotifyNewOrdersAdmin

File: src/main/java/fr/openent/crre/cron/notifyNewOrdersAdmin.java [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+
+
+    public notifyNewOrdersAdmin(ServiceFactory serviceFactory) {
+        this.notificationService = serviceFactory.getNotificationService();
+        this.vertx = serviceFactory.getVertx();
+    }
+
+    @Override
+    public void handle(Long event) {
+        log.info("CRRE notify new orders for admin cron started");
```
Comment (BastienBenard) [inline src/main/java/fr/openent/crre/cron/notifyNewOrdersAdmin.java:24]: Est ce que ça vaut le coups de logger ? Si oui String.format("[CRRE@%s::handle] Notify new orders for admin cron started", this.getClasse.getSimpleName);

File: src/main/java/fr/openent/crre/service/impl/DefaultNotificationService.java [hunk 5]
```diff
@@ -48,6 +48,21 @@ private boolean isProjectIsComplete(Map.Entry<ProjectModel, List<OrderRegionEqui
...
+                        this.serviceFactory.getOrderRegionService().getNewOrdersCount(),
+                        this.serviceFactory.getUserService().getAdminUser())
+                .onSuccess(compositeResult -> {
+                    if (((List<Neo4jUserModel>) compositeResult.resultAt(1)).size() > 0 && !((JsonObject) compositeResult.resultAt(0)).isEmpty())
+                        ((List<Neo4jUserModel>) compositeResult.resultAt(1))
+                                .forEach(adminUser -> this.prepareMessageToAdmin(adminUser.getUserId(), ((JsonObject) compositeResult.resultAt(0)).getInteger(Field.NB_ORDER.toLowerCase())));
+                })
+                .onFailure(error -> log.error(String.format("[CRRE@%s::sendNotificationAdmin] Fail to send notification to admins %s",
+                        this.getClass().getSimpleName(), error.getMessage())));
+    }
```
Comment (BastienBenard) [inline src/main/java/fr/openent/crre/service/impl/DefaultNotificationService.java:64]:
```suggestion
    @Override
    public void sendNotificationAdmin() {
        Future<JsonObject> newOrderCountFuture = this.serviceFactory.getOrderRegionService().getNewOrdersCount();
        Future<List<Neo4jUserModel>> adminUserFuture = this.serviceFactory.getUserService().getAdminUser();
        
        CompositeFuture.all(newOrderCountFuture, adminUserFuture)
                .onSuccess(compositeResult -> adminUserFuture.result()
                        .forEach(adminUser -> this.prepareMessageToAdmin(adminUser.getUserId(), newOrderCountFuture.result().getInteger(Field.NB_ORDER.toLowerCase()))))
                .onFailure(error -> log.error(String.format("[CRRE@%s::sendNotificationAdmin] Fail to send notification to admins %s",
                        this.getClass().getSimpleName(), error.getMessage())));
    }
```

File: src/main/java/fr/openent/crre/cron/notifyNewOrdersAdmin.java (hunk 3 above)
Comment (Sofianebernoussi) [reply src/main/java/fr/openent/crre/cron/notifyNewOrdersAdmin.java:12]: aucune classe n'a de maj dans le cron donc je sais pas si c'est une sorte de convention, t'es sur pour la maj?

File: src/main/java/fr/openent/crre/cron/notifyNewOrdersAdmin.java (hunk 4 above)
Comment (Sofianebernoussi) [reply src/main/java/fr/openent/crre/cron/notifyNewOrdersAdmin.java:24]: On log tout nos cron maintenant, pour avoir une trace !

File: src/main/java/fr/openent/crre/cron/notifyNewOrdersAdmin.java (hunk 3 above)
Comment (BastienBenard) [reply src/main/java/fr/openent/crre/cron/notifyNewOrdersAdmin.java]: Oui pour le coups, en java on nomme toujours les classes en CamelCase

Comment (Sofianebernoussi) [reply src/main/java/fr/openent/crre/cron/notifyNewOrdersAdmin.java]: c'est fait du coup

**Review by BastienBenard — APPROVED** [review]

**Pull request merged**